src/octoprint/static/js/app/viewmodels/terminal.js

```diff
@@ -259,10 +259,14 @@ $(function() {
 
             var re = /^([gmt][0-9]+)(\s.*)?/;
             var commandMatch = command.match(re);
+            self.blacklist =  splitTextToArray(self.settings.serial_autoUppercaseBlacklist(), ",", true);
             if (commandMatch != null) {
                 command = commandMatch[1].toUpperCase() + ((commandMatch[2] !== undefined) ? commandMatch[2] : "");
+                if (self.blacklist.indexOf(commandMatch[1].toUpperCase()) < 0){
+                    command = command.toUpperCase()
+                }
             }
-
+            
```

Comment:
Can you fix this leading whitespace? A good editior should remove it by default if you saved (e.g. Atom)